contracts/true-currencies/TrueCurrencyWithPoR.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {TrueCurrency} from "./TrueCurrency.sol";
+import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
+import {IPoRToken} from "../common/interface/IPoRToken.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+/**
+ * @title TrueCurrencyWithPoR
+ * @dev TrueCurrencyPoR is an ERC20 with blacklist & redemption addresses.
+ *  Please see TrueCurrency for the implementation that this contract inherits from.
+ *  This contract implements an additional check against a Proof-of-Reserves feed before
+ *  allowing tokens to be minted.
+ */
+abstract contract TrueCurrencyWithPoR is TrueCurrency, IPoRToken {
+    using SafeMath for uint256;
+
+    uint256 public constant MAX_CHAIN_RESERVE_HEARTBEAT = 30 days;
```

Comment:
So, I realised here https://github.com/trusttoken/contracts-pre22/pull/1176#discussion_r879425181 that we should constrain the values of `chainReserveHeartbeat` because it can potentially cause an underflow in this code:

```
require(updatedAt >= block.timestamp.sub(chainReserveHeartbeat), "TrueCurrency: PoR answer too old");
```

I think it's better to revert with a clear error message that will help in debugging rather than potentially allow the code to fail with an underflow. What do you think?

Reply:
Great idea! `updatedAt` is an answer from chainlink aggregator contract, I imagine they may be scenarios where we get a wrong value here, e.g. when owner sets `chainReserveFeed` to a wrong/malicious contract, and it's causing an underflow, however we can and probably should protect ourselves from this with something like
```
require(block.timestamp >= updatedAt)
```

Will implement this shortly, sorry about the delay, was setting up a new workstation and also found that an integration test is failing on the main branch, here is a PR with a fix: https://github.com/trusttoken/contracts-pre22/pull/1180 Can you please review and merge?